src/app/core/tools/WiringTool.ts

```diff
@@ -96,6 +125,24 @@ export const WiringTool = (() => {
 
         getWire(): Wire {
             return wire;
+        },
+
+        /**
+         * Checks for ports over objects using findPorts
+         * @param info CircuitInfo to be used by findPorts
+         * @returns whether there are any visible ports
+         */
+        visiblePorts(info: CircuitInfo): boolean {
+            return findPorts(info).length > 0;
+        },
+
+        /**
+         * To test for special case Node ports
+         * @param info CircuitInfo to be used by findPorts
+         * @returns whether there are visible ports belonging to a Node
+         */
+        nodePorts(info: CircuitInfo): boolean {
+            return findPorts(info).some(o => isNode(o.getParent()));
         }
```

Comment:
Since these are the only two functions being used from the `WiringTool` and the `findPorts` function actually doesn't depend on the `WiringTool` at all, we should instead move the `findPorts` function into a different file along with the `visiblePorts` and `nodePorts` functions and then all the files can just import from this new file instead of depending on the `WiringTool`

Specifically, we can move them into `app/core/utils/ComponentUtils.ts`